shared/config/README.md

```diff
@@ -0,0 +1,96 @@
+# Configuration Management
+
+This directory contains the centralized configuration management system for the GovDoc Scanner project.
+
+## Overview
+
+The configuration system eliminates hardcoded values throughout the codebase and provides a centralized, environment-aware configuration management solution.
+
+## Usage
+
+### Basic Import
+
+```javascript
+import { config } from "./shared/config/index.mjs";
+
+// Use configuration values
+const { baseUrl, pageLoadTimeoutMs } = config.crawler;
+const { apiKey, maxAttempts } = config.api.gemini;
+```
+
+### Configuration Validation
+
+```javascript
+import { validateConfig, validateApiKey } from "./shared/config/validator.mjs";
+
+// Validate all configuration
+try {
+  validateConfig();
+  console.log("✅ Configuration is valid");
+} catch (error) {
+  console.error("❌ Configuration Error:", error.message);
+  process.exit(1);
+}
+
+// Test API connectivity
+const apiValidation = await validateApiKey();
+if (!apiValidation.ok) {
+  console.error("❌ API Key validation failed:", apiValidation.reason);
+}
+```
+
+## Configuration Categories
+
+### API Configuration (`config.api`)
+
+Controls API-related settings, primarily for the Gemini AI service.
+
+| Variable                      | Default                   | Description               |
```

Comment:
All those variables can be injected in .env file? It would be nice to mention this